### ISSUE TITLE
Potential problem with `sentry-sdk` 1.23.0

### ISSUE THREAD

**Issue by lundberg** [kiwicom/structlog-sentry#126]
Getting this `KeyError` suddenly when `sentry-sdk` was upgraded to `1.23.0`.

```console
Traceback (most recent call last):
  File "/app/venv/lib/python3.10/site-packages/structlog_sentry/__init__.py", line 168, in _handle_event
    event, hint = self._get_event_and_hint(event_dict)
  File "/app/venv/lib/python3.10/site-packages/structlog_sentry/__init__.py", line 119, in _get_event_and_hint
    event, hint = event_from_exception(
  File "/app/venv/lib/python3.10/site-packages/sentry_sdk/utils.py", line 1038, in event_from_exception
    "values": exceptions_from_error_tuple(
  File "/app/venv/lib/python3.10/site-packages/sentry_sdk/utils.py", line 913, in exceptions_from_error_tuple
    single_exception_from_error_tuple(
  File "/app/venv/lib/python3.10/site-packages/sentry_sdk/utils.py", line 729, in single_exception_from_error_tuple
    include_source_context = client_options["include_source_context"]
KeyError: 'include_source_context'
```

Sentry SDK introduces [this line](https://github.com/getsentry/sentry-python/blob/a48a3bbc7d06fb23444b612e64d56ec6e82f1109/sentry_sdk/utils.py#L729) in [this PR](https://github.com/getsentry/sentry-python/pull/2100/files#diff-e9f88ca6368ce622839e078ed26c741a927752c01ceae64cd504d3ae86de6fd2), released in `1.23.0`.

**Comment by orangeblock** [comment]
Experiencing the same issue here, this silently fails when calling `log.exception` or `log.error(..., exc_info=True)` resulting in no logs being sent to Sentry. Downgrading to `1.22.x` works.

**Comment by LincolnPuzey** [comment]
This also effected me.

It appears to be a similar issue to https://github.com/kiwicom/structlog-sentry/issues/112
where the `client_options` dictionary supplied to `event_from_exception` does not contain keys `sentry-sdk` expects.

Is there a reason structlog-sentry constructs its own `client_options` dict, rather than just using `hub.client.options` directly which is what `sentry-sdk` appears to do internally, 
e.g. [here](https://github.com/kiwicom/structlog-sentry/blob/master/structlog_sentry/__init__.py#L114-L124) do
```
options = self._get_hub().client.options
event, hint = event_from_exception(
    exc_info,
    client_options=options,
)
```

that might prevent future issues like this from occurring.

**Comment by paveldedik** [comment]
Sorry for the slow response. I think what @LincolnPuzey is suggesting makes sense. I'll open a PR.

**Comment by paveldedik** [comment]
I hope it is fixed now (version 2.0.3). I was testing it and seems that @LincolnPuzey suggestion resolves the issue (and it also removes the need to check SDK version for compatibility - I'm kind of surprised we missed this solution from the beginning).
